cmd/zed/zed_event.c

```diff
@@ -628,6 +628,16 @@ _zed_event_add_nvpair(uint64_t eid, zed_strings_t *zsp, nvpair_t *nvp)
 	case DATA_TYPE_INT32:
 		(void) nvpair_value_int32(nvp, (int32_t *)&i32);
 		_zed_event_add_var(eid, zsp, prefix, name, "%d", i32);
+		/*
+		 * shadow readable strings for pool state
+		 */
+		if (strcmp(name, FM_EREPORT_PAYLOAD_ZFS_POOL_STATE) == 0) {
```

Comment:
Can you just add the pool state as a string with`nvlist_add_string()`, rather than making it a special case for  `DATA_TYPE_INT32`?